packages/react-components/react-popover/src/Popover.hook.ts

```diff
@@ -0,0 +1,59 @@
+import * as React from 'react';
+import { useButton } from '@react-aria/button';
+import { useOverlayPosition, useOverlayTrigger } from '@react-aria/overlays';
+import { PopoverState, UsePopoverProps } from './Popover.types';
+
+export function usePopover(state: PopoverState, props: UsePopoverProps) {
+	const {
+		maxHeight,
+		offset = 8,
+		onClose,
+		placement = 'bottom',
+		scrollRef,
+		shouldFlip = true,
+		shouldUpdatePosition = true,
+		type = 'dialog',
+	} = props;
+
+	const overlayRef = React.useRef<HTMLDivElement>(null);
+	const triggerRef = React.useRef<HTMLButtonElement>(null);
+
+	const { triggerProps, overlayProps } = useOverlayTrigger({ type }, state, triggerRef);
+	const { overlayProps: positionProps, updatePosition } = useOverlayPosition({
+		isOpen: state.isOpen,
+		maxHeight,
+		offset,
+		overlayRef,
+		placement,
+		scrollRef,
+		shouldFlip,
+		shouldUpdatePosition,
+		targetRef: triggerRef,
+	});
+
+	const { buttonProps } = useButton({ onPress: state?.toggle, ...triggerProps }, triggerRef);
```

Comment:
Dumb question, will this work with a select that has an input field?

Reply:
A consumer doesn't have to use the trigger props for the popover to work, for selects we would use it in controlled mode.